pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -162,20 +161,34 @@ func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) (*R
 	log.WithField("reapplyInterval", reapplyInterval).Info("Reapply interval set")
 	c := controllerutils.NewClientWithMetricsOrDie(mgr, ControllerName, &rateLimiter)
 	return &ReconcileClusterSync{
-		Client:                c,
-		logger:                logger,
-		reapplyInterval:       reapplyInterval,
-		resourceHelperBuilder: resourceHelperBuilderFunc,
+		Client:          c,
+		logger:          logger,
+		reapplyInterval: reapplyInterval,
 		remoteClusterAPIClientBuilder: func(cd *hivev1.ClusterDeployment) remoteclient.Builder {
 			return remoteclient.NewBuilder(c, cd, ControllerName)
 		},
+		resourceHelperBuilder: resourceHelperBuilderFunc,
 	}, nil
 }
 
-func resourceHelperBuilderFunc(restConfig *rest.Config, fakeCluster bool, logger log.FieldLogger) (resource.Helper, error) {
-	if fakeCluster {
+func resourceHelperBuilderFunc(
+	cd *hivev1.ClusterDeployment,
+	remoteClusterAPIClientBuilderFunc func(cd *hivev1.ClusterDeployment) remoteclient.Builder,
+	logger log.FieldLogger,
+) (
+	resource.Helper,
+	error,
+) {
+	if controllerutils.IsFakeCluster(cd) {
 		return resource.NewFakeHelper(logger), nil
 	}
+
+	restConfig, err := remoteClusterAPIClientBuilderFunc(cd).RESTConfig()
```

Comment:
Rest config moved to here within `resourceHelperBuilderFunc`. Tests no longer need to mock a call to `.RESTConfig()` since the test helper returns a fake resource helper.